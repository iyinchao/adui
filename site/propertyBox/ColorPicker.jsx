import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Button, Switch, ColorPicker, Radio } from "../../components"
import { SIZES } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    disabled: false,
    inputVisible: true,
    size: "small",
  }

  render() {
    const { disabled, inputVisible, size } = this.state

    const codes = `<ColorPicker
  defaultValue="#07C160"
  ${disabled ? `disabled` : ""}
  ${!inputVisible ? `inputVisible={false}` : ""}
  ${size === "small" ? "" : `size="${size}"`}
/>`
    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <ColorPicker
              defaultValue="#07C160"
              disabled={disabled}
              inputVisible={inputVisible}
              size={size}
            />
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
            </dl>
            <dl className={styles.controls}>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="禁用态"
                  unCheckedText="禁用态"
                  checked={disabled}
                  onChange={(param) => this.setState({ disabled: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="带输入框"
                  unCheckedText="带输入框"
                  checked={inputVisible}
                  onChange={(param) => this.setState({ inputVisible: param })}
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
