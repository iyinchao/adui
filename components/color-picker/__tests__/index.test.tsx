/* eslint-disable */
import React, { Component } from "react"
import { mount } from "enzyme"
import TestRenderer from "react-test-renderer"
// import { act } from "react-dom/test-utils"
import ColorPicker from ".."
import { act } from "react-dom/test-utils"

class ColorPickerMounter extends Component {
  state = {
    value: "#07C160",
  }

  render() {
    const { value } = this.state
    return (
      <ColorPicker
        value={value}
        onChange={(val) => this.setState({ value: val })}
        {...this.props}
      />
    )
  }
}

describe("ColorPicker", () => {
  beforeAll(() => {
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  it("是否正确渲染", () => {
    const wrapper = TestRenderer.create(<ColorPicker defaultValue="#07C160" />)
    expect(wrapper).toMatchSnapshot()
  })

  it("是否能处理小写值的情况", () => {
    const wrapper = mount(<ColorPicker defaultValue="#07c160" />)
    const value = wrapper.find(".adui-cp").at(0).props()["data-value"]
    const inputValue = wrapper.find("Input").at(0).props().value
    expect(inputValue === "07C160" && value === "#07C160").toBe(true)
  })

  it("内部驱动", () => {
    const wrapper = mount(<ColorPicker defaultValue="#07C160" />)
    wrapper.find("button").simulate("click")
    wrapper.find(".adui-cp-standard i").at(0).simulate("click")
    const value = wrapper.find(".adui-cp").at(0).props()["data-value"]
    const inputValue = wrapper.find("Input").at(0).props().value
    expect(inputValue === "FFFFFF" && value === "#FFFFFF").toBe(true)

    wrapper
      .find("input")
      .at(0)
      .simulate("change", { target: { value: "F0F0F0" } })

    const value1 = wrapper.find(".adui-cp").at(0).props()["data-value"]
    const inputValue1 = wrapper.find("Input").at(0).props().value
    expect(inputValue1 === "F0F0F0" && value1 === "#F0F0F0").toBe(true)
  })

  it("外部控制", () => {
    const wrapper = mount(<ColorPickerMounter />)

    wrapper.find("button").simulate("click")
    act(() => {
      wrapper.find(".adui-cp-standard i").at(0).simulate("click")
      jest.runAllTimers()
      wrapper.update()
      const value = wrapper.find(".adui-cp").at(0).props()["data-value"]
      const inputValue = wrapper.find("Input").at(0).props().value
      expect(
        inputValue === "FFFFFF" &&
          value === "#FFFFFF" &&
          wrapper.state().value === "#FFFFFF"
      ).toBe(true)
    })
    wrapper
      .find("input")
      .at(0)
      .simulate("change", { target: { value: "F0F0F0" } })
    const value1 = wrapper.find(".adui-cp").at(0).props()["data-value"]
    const inputValue1 = wrapper.find("Input").at(0).props().value
    expect(
      inputValue1 === "F0F0F0" &&
        value1 === "#F0F0F0" &&
        wrapper.state().value === "#F0F0F0"
    ).toBe(true)
  })

  it("禁用态", () => {
    const wrapper = mount(<ColorPicker defaultValue="#07C160" disabled />)
    wrapper.find("button").simulate("click")
    expect(wrapper.find("Popover").props().visible).toBe(false)
  })

  it("popupResultVisible", () => {
    const wrapper0 = mount(<ColorPicker defaultValue="#07C160" />)
    const wrapper1 = mount(
      <ColorPicker defaultValue="#07C160" popupResultVisible={false} />
    )
    wrapper0.find("button").simulate("click")
    wrapper1.find("button").simulate("click")
    expect(wrapper0.find("input")).toHaveLength(2)
    expect(wrapper1.find("input")).toHaveLength(1)
  })

  it("handlePrefixClick 验证", () => {
    const wrapper = mount(<ColorPicker defaultValue="#07C160" />)
    wrapper.find(".adui-cp-prefix").simulate("click")
    wrapper.find("button").simulate("click")
    jest.advanceTimersByTime(50)
    wrapper.find(".adui-cp-prefix").at(1).simulate("click")
  })
})
