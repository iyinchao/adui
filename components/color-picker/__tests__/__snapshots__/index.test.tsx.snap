// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker 是否正确渲染 1`] = `
<div
  className="adui-cp adui-cp-small"
  data-b={75.68627450980392}
  data-h={148.70967741935485}
  data-s={96.37305699481865}
  data-value="#07C160"
>
  <button
    className="adui-cp-picker adui-button-base adui-button-normal adui-button-small"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="adui-button-content"
    >
      <i
        style={
          Object {
            "background": "#07C160",
          }
        }
      />
    </span>
  </button>
  <div
    className="adui-cp-input adui-input-wrapper adui-input-small adui-input-normal"
    style={Object {}}
  >
    <div
      className="adui-input-left"
      onClick={[Function]}
      role="none"
    >
      <div
        className="adui-cp-prefix"
        onClick={[Function]}
        role="none"
      >
        #
      </div>
    </div>
    <input
      autoComplete="off"
      autoFocus={false}
      className="adui-input-base"
      disabled={false}
      id={null}
      name={null}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      required={false}
      style={
        Object {
          "paddingLeft": undefined,
          "paddingRight": undefined,
        }
      }
      type={null}
      value="07C160"
    />
  </div>
</div>
`;
